flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -953,7 +954,7 @@ private static CatalogPartitionSpec createPartitionSpec(String hivePartitionName
 		Map<String, String> spec = new HashMap<>(partKeyVals.length);
 		for (String keyVal : partKeyVals) {
 			String[] kv = keyVal.split("=");
-			spec.put(kv[0], kv[1]);
+			spec.put(unescapePathName(kv[0]), unescapePathName(kv[1]));
```

Comment:
Not related to this PR, but maybe we need more tests for this, e.g. string partition columns with special characters.

Reply:
I added test in `HiveTableSourceTest.testPartitionFilter`